frontend/src/components/organisms/ProfileForm.tsx

```diff
@@ -39,8 +41,76 @@ interface ProfileFormProps {
 
 const ProfileForm = ({ userDetails }: ProfileFormProps) => {
   const router = useRouter();
+  const [updatePassword, updating, error] = useUpdatePassword(auth);
+  const [success, setSuccess] = React.useState<boolean>(false);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = React.useState<string>("");
 
-  // Future testing: What if user doesn't have a nickname, does the code break?
+  const handleErrors = (errors: any) => {
+    const errorParsed = errors?.split("/")[1].slice(0, -2);
+    switch (errorParsed) {
+      case "invalid-email":
+        return "Invalid email address format.";
+      case "user-disabled":
+        return "User with this email has been disabled.";
+      case "user-not-found":
+        return "There is no user with this email address.";
+      case "wrong-password":
+        return "Old password is incorrect.";
+      case "weak-password":
+        return "Password must be at least 6 characters.";
+      case "invalid-password":
+        return "Invalid password.";
+      case "requires-recent-login":
+        return "Please reauthenticate to change your password.";
+      case "too-many-requests":
+        return "You have made too many requests to change your password. Please try again later.";
+      default:
+        return "Something went wrong.";
+    }
+  };
+
+  const ProfileErrorComponent = (): JSX.Element => {
+    return (
+      <div>
+        {error ? (
+          <Alert
+            severity="error"
+            title="Error"
+            message={handleErrors(error?.message)}
+          />
+        ) : null}
+      </div>
+    );
+  };
+
+  const ProfileErrorFromReauthenticating = (): JSX.Element => {
+    return (
+      <div>
+        {errorMessage.length > 0 ? (
+          <Alert
+            severity="error"
+            title="Error"
+            message={handleErrors(errorMessage)}
+          />
+        ) : null}
+      </div>
+    );
+  };
+
+  const ProfileSuccessComponent = (): JSX.Element => {
```

Comment:
I think we should either put this micro success / fail components in atoms or their own folder just to reduce the clutter

Reply:
Definitely there does seem to be a lot of duplication but putting them in atoms isn't the right solution; atoms should be for generic reusable components with no business logic tied to them, and in this case, all of the micro components are just wrappers around the existing `Alert` atom. Will look into different ways to refactor this

Reply:
Maybe could be in utils or similar